tt_metal/impl/dispatch/kernels/cq_prefetch.cpp

```diff
@@ -184,7 +197,10 @@ void fetch_q_get_cmds(uint32_t& fence, uint32_t& cmd_ptr, uint32_t& pcie_read_pt
     }
 
     bool cmd_ready = (cmd_ptr != fence);
-    uint32_t fetch_size = (uint32_t)*prefetch_q_rd_ptr << prefetch_q_log_minsize;
+
+    uint32_t fetch_size = ((uint32_t)(*prefetch_q_rd_ptr) & ~prefetch_q_msb_mask) << prefetch_q_log_minsize;
+    bool stall_flag = (*prefetch_q_rd_ptr & prefetch_q_msb_mask) != 0;
```

Comment:
prefetch_q_rd_ptr is a volatile L1 address, only read it once (and store result in a local register)

Reply:
Good point..thanks!  This seems like the right pattern (applied to both places in code).

```
    uint32_t prefetch_q_rd_ptr_local = *prefetch_q_rd_ptr;
    uint32_t fetch_size = (prefetch_q_rd_ptr_local & ~prefetch_q_msb_mask) << prefetch_q_log_minsize;
    bool stall_flag = (prefetch_q_rd_ptr_local & prefetch_q_msb_mask) != 0;```